content/blog/parsnip-update-03-2022/index.Rmd

```diff
@@ -0,0 +1,139 @@
+
+A good overview of this model is: _Bayesian Additive Regression Trees: A Review and Look Forward_ ([pdf](https://par.nsf.gov/servlets/purl/10181031)). 
+
+### New engines
+
+Within parsnip, a `"glm"` engine was added for linear regression. An engine vale of `"brulee"` was added for linear, logistic, and multinomial regression as well as for neural networks. brulee is a new package for fitting models using torch (look for a blog post soon on this package). 
+
+As discussed below, the multilevelmod package adds a lot more engines for linear(ish) models, such as [`"gee"`](https://parsnip.tidymodels.org/reference/details_linear_reg_gee.html), [`"gls"`](https://parsnip.tidymodels.org/reference/details_linear_reg_gls.html), [`"lme"`](https://parsnip.tidymodels.org/reference/details_linear_reg_lme.html), [`"lmer"`](https://parsnip.tidymodels.org/reference/details_linear_reg_lmer.html), and [`"stan_glmer"`](https://parsnip.tidymodels.org/reference/details_linear_reg_stan_glmer.html). There are similar engines for logistic and Poisson regression. 
+
+
+## multilevelmod
+
+This package has been simmering for a while on GitHub. Its engines are useful for fitting a variety of models that go by a litany of different names: mixed effects models, random coefficient models, variance component models, hierarchical linear models, and so on. 
+
+One aspect of these models is that they mostly work with the formula method, which specifies both the model terms and also which of these are "random effects". 
+
+As an example, let's look at the measurement system analysis (MSA) data in the package. In these data, 56 separate items were measured twice using a laboratory test. The lab would like to understand how noisy their data are and if different samples can be distinguished from one another. Here's a plot of the data:  
+
+```{r data-plot, fig.align='center', fig.width=9, fig.height=4, dev = "svg"}
+library(ggplot2)
+library(parsnip)
+library(multilevelmod)
+
+data(msa_data)
+
+msa_data %>% 
+  ggplot() + 
+  aes(x = reorder(id, value), y = value, col = replicate, pch = replicate) + 
+  geom_point(alpha = 1/2, cex = 3) + 
+  labs(x = NULL, y = "lab result") +
+  theme_bw() + 
+  theme(
+    axis.text.x = element_text(angle = 90), 
+    legend.position = "top"
+  )
+```
+
+
+In this data set, the goal is to estimate how much of the variation in the lab test is due to the different samples (as it should be since they are different) or measurement noise. The latter term could be associated with day-to-day differences, people-to-people differences etc. It might also be irreducible noise. In any case, we'd like to get estimates of these two sources of variation. 
```

Comment:
```suggestion
With this data set, the goal is to estimate how much of the variation in the lab test is due to the different samples (as it should be since they are different) or measurement noise. The latter term could be associated with day-to-day differences, people-to-people differences etc. It might also be irreducible noise. In any case, we'd like to get estimates of these two sources of variation. 
```